src/zenml/metadata_stores/mysql_metadata_store.py

```diff
@@ -37,11 +42,93 @@ def get_tfx_metadata_config(
         metadata_store_pb2.ConnectionConfig,
         metadata_store_pb2.MetadataStoreClientConfig,
     ]:
-        """Return tfx metadata config for mysql metadata store."""
-        return metadata.mysql_metadata_connection_config(
+        """Return tfx metadata config for MySQL metadata store."""
+        config = MySQLDatabaseConfig(
             host=self.host,
             port=self.port,
             database=self.database,
-            username=self.username,
-            password=self.password,
         )
+
+        secret = self._get_mysql_secret()
+
+        # Set the user
+        if self.user:
+            if secret and secret.user:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a username `{self.user}` and `{secret.user}`. Please "
+                    f"make sure that you only use one."
+                )
+            else:
+                config.user = self.user
+        else:
+            if secret and secret.user:
+                config.user = self.user
+            else:
+                raise RuntimeError(
+                    "Your metadata store does not have a username. Please "
+                    "provide it either by defining it upon registration or "
+                    "through a MySQL secret."
+                )
+
+        # Set the password
+        if self.password:
+            if secret and secret.password:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a password. Please make sure that you only use one."
+                )
+            else:
+                config.password = self.password
+        else:
+            if secret and secret.user:
+                config.password = self.password
+
+        # Set the SSL configuration if there is one
+        if secret:
+            ssl_options = {}
+            # Handle the files
+            for key in ["ssl_key", "ssl_ca", "ssl_cert"]:
+                content = getattr(secret, key)
+                filepath = os.path.join(os.getcwd(), f"{key}.pem")
+                ssl_options[key.lstrip("ssl_")] = filepath
+                with open(filepath, "w") as f:
+                    f.write(content)
```

Comment:
I would also recommend that you set minimum permissions for the file(s) (ox600 or rw-------) after creation, especially for the private client certificate key, if running on a filesystem and OS where such a thing is possible. Also for the parent folder, if created by ZenML, I would recommend setting the minimum permissions (ox700 or rwx------). These precautions are standard Linux permissions for sensitive files such as private certificate keys. Some even go as far as removing all write permissions for the file.

Reply:
Nice suggestion! Implemented it.